sdk/eventhub/event-hubs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,256 @@
+
+### Connect using an IoT connection string
+
+Because translating a connection string requires querying the IoT Hub service, the Event Hubs client library cannot use it directly. The [iothubConnectionString.js][iothubconnectionstring] sample describes how to query IoT Hub to translate an IoT connection string into one that can be used with Event Hubs.
+
+Further reading:
+
+- [Control access to IoT Hub using Shared Access Signatures][iothubsas]
+- [Read device-to-cloud messages from the built-in endpoint][ioteventhubendpoint]
+
+### Cannot add components to the connection string
+
+The legacy Event Hub clients allowed customers to add components to the connection string retrieved from the portal. The legacy clients are in packages [@azure/event-hubs@2.0.0][event_hubs_v2] and [@azure/event-processor-host][event_processor_host]. The current generation supports connection strings only in the form published by the Azure portal.
+
+#### Adding "TransportType=AmqpWebSockets"
+
+To use web sockets, see the sample [websockets.js][publisheventswithwebsockets].
+
+#### Adding "Authentication=Managed Identity"
+
+To authenticate with Managed Identity, see the sample [usingAadAuth.js][publisheventswithazureidentity].
+
+For more information about the `Azure.Identity` library, check out our [Authentication and the Azure SDK][authenticationandtheazuresdk] blog post.
+
+## Enable and configure logging
+
+The Azure SDK for JavaScript offers a consistent logging story to help troubleshoot application errors and expedite their resolution. The logs produced will capture the flow of an application before reaching the terminal state to help locate the root issue.
+
+You can set the `AZURE_LOG_LEVEL` environment variable to enable logging to `stderr`:
+
+```bash
+export AZURE_LOG_LEVEL=verbose
+```
+
+For more detailed instructions on how to enable logs, you can look at the
+[@azure/logger package docs](https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/core/logger).
+
+### Enable AMQP transport logging
+
+You can alternatively set the `DEBUG` environment variable to get logs when using this library.
+This can be useful if you also want to emit logs from the dependencies `rhea-promise` and `rhea` as well.
+
+**Note:** AZURE_LOG_LEVEL, if set, takes precedence over DEBUG.
+Do not specify any `azure` libraries via DEBUG when also specifying
+AZURE_LOG_LEVEL or calling setLogLevel.
+
+- Getting only info level debug logs from the Event Hubs client library.
+
+```bash
+export DEBUG=azure:*:info
+```
+
+- Getting debug logs from the Event Hubs client library and the protocol level library.
+
+```bash
+export DEBUG=azure*,rhea*
+```
+
```

Comment:
it would be nice to have information about configuring DEBUG when running in browser

To enable DEBUG logging for browser using `localStorage.debug`

```js
localStorage.debug = "azure:*:info";
```
Note that this is persisted in browser local storage, so to disabled it. Set it back to `undefined`
```js
localStorage.debug = undefined;
```

Reply:
Good point. However, it is my understanding that event hubs clients shouldn't be used in the browsers and/or similar clients but I could be wrong. I added it for now.